site/packages/aa-alchemy/provider/withAlchemyEnhancedApis.md

```diff
@@ -0,0 +1,52 @@
+---
+outline: deep
+head:
+  - - meta
+    - property: og:title
+      content: AlchemyProvider • withAlchemyEnhancedApis
+  - - meta
+    - name: description
+      content: Overview of the withAlchemyEnhancedApis method on Alchemy Provider in aa-alchemy
+  - - meta
+    - property: og:description
+      content: Overview of the withAlchemyEnhancedApis method on Alchemy Provider in aa-alchemy
+---
+
+# withAlchemyEnhancedApis
```

Comment:
I missed to bring this up from the previous PR where Enhanced APIs functionality is implemented.

wdyt about on the base `SmartAccountProvider`, having `withEnhancedApis` where it takes an **interface** defining methods for fetching balance, erc20s, NFTs, simulation, etc. **selectively** based on the customer's need? 

My concern is that when we want to have other enhanced api functionality that does not involve Alchemy SDK, or want to integrate account kit on a new chain, we are not blocked/dependent on Alchemy SDK being updated.

Also, while not enforcing the usage of Alchemy SDK at the aa-core level, we still gives the flexibility for devs to enjoy our cleaner Account Kit usage experience.

thoughts? @avasisht23 @moldy530

Reply:
hmm I'd rather have the e-apis on the alchemy provider.
1. upsell to use alchemy specific apis on alchemy provider (and by convenience use other alchemy aa stuff already in the same library i.e. gas manager, bundler, light account, etc.)
2. general principle of core being a building blocks repository, and alchemy being opinionated on using alchemy tech for convenience
3. interface on core might be overkill. the interface would require us to re-define types that alchemy-sdk gives us for free, which is a dupe and requires maintenance from us to sync it with alchemy-sdk changes anyways (as far as i understand your idea).

feel free to put up a branch with your vision here - I might not be seeing it correctly. that said, think it's lower pri than other stuff we're working on :)